src/domain/entities/OptionValidationResult.ts

```diff
@@ -0,0 +1,15 @@
+import { Option } from "./Option";
+import { OptionSet } from "./OptionSet";
+
+type RuleType = "duplicate" | "naming_conventions" | "invalid_length" | "misplaced_commas" | "order";
+
+export type OptionValidationResult = {
+    option: Option;
+    optionSet: Omit<OptionSet, "options">;
+    rules: OptionValidationRule[];
+};
+
+export type OptionValidationRule = {
```

Comment:
a "rule" or "validationRule" conveys the idea of definition, but here we are describing only problems, right? let's add that to prop "rules" and name "OptionValidationRule" (error? unmet? invalid?)